pkg/engines/tekton/utils.go

```diff
@@ -0,0 +1,256 @@
+package tekton
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/jenkins-x/lighthouse/pkg/apis/lighthouse/v1alpha1"
+	"github.com/jenkins-x/lighthouse/pkg/config"
+	"github.com/jenkins-x/lighthouse/pkg/jobutil"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tektonclient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	controllerName          = "tekton-controller"
+	gitCloneCatalogTaskName = "git-clone"
+	gitCloneURLParam        = "url"
+	gitCloneRevisionParam   = "revision"
+	gitMergeCatalogTaskName = "git-batch-merge"
+	gitMergeBatchRefsParam  = "batchedRefs"
+)
+
+var reProwExtraRef = regexp.MustCompile(`PROW_EXTRA_GIT_REF_(\d+)`)
```

Comment:
@abayer do you plan on getting this magic stuff work in lighthouse ?
It looks like it's used to validate the pipeline run but is never applied.
If it's ok i can remove it :D

Reply:
I'd say remove it. I never quite figured out what the point was anyway. =)

Reply:
done